src/couch/src/couch_users_db.erl

```diff
@@ -86,7 +90,63 @@ save_doc(#doc{body={Body}} = Doc) ->
         Doc#doc{body={Body4}};
     {_ClearPassword, Scheme} ->
         couch_log:error("[couch_httpd_auth] password_scheme value of '~p' is invalid.", [Scheme]),
-        throw({forbidden, "Server cannot hash passwords at this time."})
+        throw({forbidden, ?PASSWORD_SERVER_ERROR})
+    end.
+
+% Validate if a new password matches all RegExp in the password_reqexp setting.
+% Throws if not.
+validate_password(ClearPassword) ->
+    case config:get("couch_httpd_auth", "password_reqexp", "") of
+    "" ->
+        ok;
+    "[]" ->
+        ok;
+    ValidateConfig ->
+        case couch_util:parse_term(ValidateConfig) of
+        {ok, RegExpList} when is_list(RegExpList) -> 
+            % Check the password on every RegExp.
+            Loop = fun(RegExp) ->
+                check_password_with_regexp(ClearPassword, RegExp)
+            end,
+            lists:foreach(Loop, RegExpList),
+            ok;
+        {ok, NonListValue} ->
+            couch_log:error(
+                "[couch_httpd_auth] password_reqexp value of '~p' is invalid.",
```

Comment:
`?MODULE` is preferred to hard-coding the module name here. Also, please omit the `[]`

Reply:
This is the configuration name, inspired by the error message of [`save_doc/1`](https://github.com/apache/couchdb/blob/776f920021dd15c51355e9a904bdc5a2250b19d9/src/couch/src/couch_users_db.erl#L88).

Reply:
Is your request to change the error logs to be more like [this](https://github.com/apache/couchdb/blob/776f920021dd15c51355e9a904bdc5a2250b19d9/src/mem3/src/mem3_rep.erl#L496)
Like this so:
`couch_log:error("~p : ~p couldn't be parsed. '~p' is not a list.", [?MODULE, "password_regexp", NonListValue])`

Reply:
@jaydoane if another auth module supports PW validation it can hardcode its own name, so I think this is fine here. — And I prefer the convention of `[section] key` for referring to config value.